src/main/java/org/opentripplanner/routing/core/FareRuleSet.java

```diff
@@ -33,10 +37,22 @@ public Set<P2<String>> getOriginDestinations() {
     return originDestinations;
   }
 
+  public void addRouteOriginDestination(String route, String origin, String destination) {
+    routeOriginDestinations.add(new P3<String>(route, origin, destination));
```

Comment:
I'd think it is better to have a named record for these

Reply:
What Hannes means by this is that you should not create the `P3` type but a named record like this:

```
public record RouteOriginDestination(String route, String origin, String destination) {}
```

That way you can avoid the confusing calls to `first`, `second` and `third` as it's arbitrary what that means.

Reply:
Ok, changed P3 into a named record